index.js

```diff
@@ -220,12 +248,50 @@ function mine() {
 
 function fluctuate() {
     const priceSave = price;
+    var goldTotal = 0;
+    var meatTotal = 0;
+    //var currentTime = new Date().getTime() ;
+    //var timeSinceOrigin = (parseFloat(originTime) - currentTime)/-1000;
+    var userData;
+    if (Object.keys(ledger).length > 0)
+        Object.keys(ledger).forEach(function(key) {
+            userData = ledger[key];
+            goldTotal += parseFloat(userData.gold.toFixed(2));
+            meatTotal += parseFloat(userData.meatCoin.toFixed(2));
+          });
+    //Adjust by random fluctuations, trending down for deflation:
     const priceAdjust = 1 + Math.random()/75.0;
-    if (Math.random() > 0.5)
+    var adjuster = Math.random();
+    var hyperAdjuster;
+    if (adjuster < 0.6)
         price /= priceAdjust;
-    else
+    //Rare chance of large "hyper" fluctuation
+    else if (adjuster > .99) {
+        hyperAdjuster = 1 + Math.random()/2;
+        console.log("HYPER! " + hyperAdjuster);
+        if (Math.random() > 0.5) {
+            price /= hyperAdjuster;
+        } else {
+            price *= hyperAdjuster;
+        }
+    } else {
         price *= priceAdjust;
+    }
+    //Adjust for total market, scaled down by 7:
+    if (meatTotal > 0) {
+      price = price*((7*meatTotal+meatTotal)/(7*meatTotal+meatTotalLast));
```

Comment:
Can you explain how this works?

Reply:
It basically is a way of reducing how much the market scales. But in a nutshell, as the amount of meat increases since the last adjustment, the market will adjust proportionately. Changing the 7 changes how dramatically the market will adjust. Increasing it decreases the volatility.

Reply:
Isn't 7*meatTotal+meatTotal just 8*meatTotal. Why 7?

Reply:
the denominator is meatTotalLast, not meat Total. the equation just scales with the 7, easy way to adjust in case its too fast or too slow.

Reply:
The numerator is 7*meatTotal+meatTotal.